sylt-common/src/error.rs

```diff
@@ -281,3 +279,97 @@ impl fmt::Display for RuntimeError {
         }
     }
 }
+
+#[cfg(test)]
+mod test {
+    // A small hack is required to test the functions working on Formatters
+    // since we can't construct new Formatters.
+    //
+    // The formatted span output is very weird to test. Feel free to move the
+    // string around and check that the ^^^ lines up correctly. Spaces matter!
+    //
+    // For some tests, color is disabled by setting the env variable NO_COLOR=1.
+
+    struct UnderlineTester(usize, usize);
+    impl std::fmt::Display for UnderlineTester {
+        fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+            super::underline(f, self.0, self.1)
+        }
+    }
+
+    struct SourceSpanTester<'p>(&'p std::path::Path, super::Span);
+    impl<'p> std::fmt::Display for SourceSpanTester<'p> {
+        fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+            super::write_source_span_at(f, self.0, self.1)
+        }
+    }
+
+    #[test]
+    fn underline() {
+        assert_eq!(&format!("{}", UnderlineTester(1, 2)), " ^^\n");
+        assert_eq!(&format!("{}", UnderlineTester(2, 1)), "  ^\n");
+        assert_eq!(&format!("{}", UnderlineTester(0, 0)), "\n");
+        assert_eq!(&format!("{}", UnderlineTester(0, 2)), "^^\n");
+    }
+
+    fn get_tmp() -> std::path::PathBuf {
+        use rand::Rng;
```

Comment:
...
https://crates.io/crates/sungod

Do we even need rand to begin with? Why not give in a funky string that makes the files unique?

Reply:
The funky string has to be unique for every test and a random number felt like the easiest. But I forgot the sungod existed. All hail!